### PR TITLE
Vertex cover

### PR DESCRIPTION
**Pull request by olegfafurin**
### Update

Added **minimum** vertex cover algorithm formulated as JuMP optimization problem. 
- implementation is for **undirected** graphs only (as it's not clear what's a cover in a directed graph case)
- finds **some** vertex cover in a graph
- returns a vector of vertices

### CODE REVIEW

File: docs/src/algorithms.md [hunk 1]
```diff
@@ -61,6 +61,14 @@ min_cost_assignment
 GraphsOptim.min_cost_assignment!
 ```
 
+## Minimum Vertex Cover
+
+```@docs
+min_vertex_cover
```
Comment (gdalle) [inline docs/src/algorithms.md:67]: Apparently there is a missing docstring, which is why the docs doesn't build

File: src/vertex_cover.jl [hunk 2]
```diff
@@ -0,0 +1,68 @@
+"""
+    vertex_cover!(
```
Comment (gdalle) [inline src/vertex_cover.jl]: This name must align with the name of the function

File: src/vertex_cover.jl [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+"""
+function min_vertex_cover!(model::Model, g::AbstractGraph; var_name)
+    if is_directed(g)
+        throw(ArgumentError("The graph must not be directed"))
+    end
+
+    ver = collect(vertices(g))
+    edge_tuples = [(src(ed), dst(ed)) for ed in edges(g)]
+
+    f = @variable(model, [ver]; integer=true, base_name=String(var_name))
```
Comment (gdalle) [inline src/vertex_cover.jl]: You can directly declare bounds in the variable creation

Comment (gdalle) [reply src/vertex_cover.jl]: Also it would be nice to leave the option `integer` to the user

File: src/vertex_cover.jl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    model[Symbol(var_name)] = f
+
+    for v in ver
+        @constraint(model, f[v] >= 0)
+        @constraint(model, f[v] <= 1)
+    end
+
+    for (u, v) in edge_tuples
```
Comment (gdalle) [inline src/vertex_cover.jl]: You don't need to pre-allocate the tuples, you can just iterate over the edges here (microscopic performance win)

File: src/vertex_cover.jl [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+function min_vertex_cover(g::AbstractGraph; optimizer=HiGHS.Optimizer)
+    model = Model(optimizer)
+    set_silent(model)
+
+    min_vertex_cover!(model, g; var_name=:cover)
+
+    optimize!(model)
+    @assert termination_status(model) == OPTIMAL
+
+    cover_values = Vector(value.(model[:cover]))
```
Comment (gdalle) [inline src/vertex_cover.jl]: `value.` will already return a `Vector` here, no need for the explicit conversion

File: test/vertex_cover.jl [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+adj_matrix = [
+    0 1 1 0 0 0
+    1 0 1 1 1 1
+    1 1 0 0 0 0
+    0 1 0 0 0 0
+    0 1 0 0 0 0
+    0 1 0 0 0 0
+]
+
+expected_ans_options = Set([Set([1, 2]), Set([2, 3])])
```
Comment (gdalle) [inline test/vertex_cover.jl]: You don't need the outer `Set` here

**Review by gdalle — COMMENTED** [review]
Good work! A few minor kinks and we can merge it :) Congrats on your first contribution to the graphs ecosystem

File: src/vertex_cover.jl (hunk 5 above)
Comment (olegfafurin) [reply src/vertex_cover.jl]: It returns a `JuMP.Containers.DenseAxisArray` so I guess I should unpack the data somehow (either with `.data` or like this?)

File: src/vertex_cover.jl (hunk 3 above)
Comment (olegfafurin) [reply src/vertex_cover.jl]: > You can directly declare bounds in the variable creation

Not with the anonymous variables afaik

File: src/vertex_cover.jl (hunk 5 above)
Comment (gdalle) [reply src/vertex_cover.jl]: Fine by me

**Review by gdalle — APPROVED** [review]

**Pull request merged**